packages/components/src/hooks/useImage.ts

```diff
@@ -0,0 +1,91 @@
+"use client"
+
+import { useState } from "react"
+
+const waitFor = (baseTimeoutMs: number = 500) => {
+  return new Promise((resolve) => setTimeout(resolve, baseTimeoutMs))
+}
+
+const retry = async (
+  promise: () => Promise<string>,
+  baseTimeoutMs: number,
+  maxRetries: number,
+) => {
+  const retryWithBackoff = async (retries: number) => {
+    try {
+      // Make sure we don't wait on the first attempt
+      if (retries > 0) {
+        const timeToWait = baseTimeoutMs * 2 ** retries
+        console.log(`waiting for ${timeToWait}ms...`)
```

Comment:
issue(non-blocking): hmm this console.log will appear on the user's browser (since this is client-side), I think we should remove?

Reply:
sounds good